source/dub/commandline.d

```diff
@@ -257,6 +257,7 @@ struct CommonOptions {
 	/// Parses all common options and stores the result in the struct instance.
 	void prepare(CommandArgs args)
 	{
+		registry_urls = environment.get("DUB_REGISTRY", null).split(";");
```

Comment:
It's weird to pass package suppliers to `new Dub` only to have it add more package suppliers internally.
Also the env variables are simple alternatives to the command line options, so treating them somewhere else doesn't seem too sensible.